crates/vm2/src/world_diff.rs

```diff
@@ -290,7 +277,7 @@ impl WorldDiff {
         }
     }
 
-    pub(crate) fn rollback(&mut self, snapshot: Snapshot) {
+    pub(crate) fn rollback(&mut self, snapshot: &Snapshot) {
```

Comment:
The Snapshot should be consumed. Rolling back to it multiple times isn't intended use.

Reply:
Same with ExternalSnapshot